src/main/java/com/truelayer/java/TrueLayerClient.java

```diff
@@ -64,6 +66,17 @@ public IPaymentsApi payments() {
         return paymentsHandler;
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public IPaymentsProvidersHandler paymentsProviders() {
+        if (ObjectUtils.isEmpty(paymentsProvidersHandler)) {
+            throw buildInitializationException("payments providers");
+        }
```

Comment:
We could probably remove this block considering how the client is created: payments, merchant and mandates handlers might be empty if the `TrueLayerClient` instance is built without signature options.

Or, adopt the same approach inside the `auth()` method above